kentikapi/internal/cloud/export.go

```diff
@@ -304,66 +308,62 @@ func validateCEProvider(ce *cloud.Export) error {
 	case ibmProvider:
 		return validateIBMProvider(ce)
 	default:
-		return kentikerrors.New(
-			kentikerrors.InvalidRequest,
-			fmt.Sprintf("cloud provider '%s' is not supported", ce.Provider))
+		return fmt.Errorf("cloud provider '%s' is not supported", ce.Provider)
 	}
 }
 
 func validateAWSProvider(ce *cloud.Export) error {
```

Comment:
Could this repetitive code be possibly generalized or meta-programmed (i.e. generated from some template)?

Reply:
Yes, it is possible.
1. We could write a function using reflections that would check required fields based on struct tags. We have something similar to that for checking nil pointers in API v5 request/response payloads: 
https://github.com/kentik/community_sdk_golang/blob/d23c4376f09d48d30357550569d0298f029081e7/kentikapi/internal/validation/validate.go#L18
It, however does not check for zero-values (only nil pointers).
2. Use a library for validation that also uses struct tags: https://github.com/go-playground/validator / https://github.com/asaskevich/govalidator
3. Use a library for validation that do not use struct tags: https://github.com/go-ozzo/ozzo-validation/

Reply:
I added a note about that to synthetics validation task: https://clkentik.atlassian.net/browse/KNTK-402